code/modules/economy/account.dm

```diff
@@ -13,18 +13,36 @@
 	var/bounty_timer = 0
 
 /datum/bank_account/New(newname, job, modifier = 1)
-	if(add_to_accounts)
-		SSeconomy.bank_accounts += src
 	account_holder = newname
 	account_job = job
 	account_id = rand(111111,999999)
 	payday_modifier = modifier
+	if(add_to_accounts)
+		var/iterations = 100
+		while(SSeconomy.bank_accounts_by_id["[account_id]"] && iterations) //still more likely than winning a pulse rifle at the arcade
+			account_id++
+			iterations--
+			if(account_id > 999999)
+				account_id = 111111
```

Comment:
```suggestion
		for(var/i in 1 to 100)
			if(!SSeconomy.bank_accounts_by_id["[account_id]"])
				break
			account_id++ //still more likely than winning a pulse rifle at the arcade
			if(account_id > 999999)
				account_id = 111111
```

Reply:
Could you turn this into a `get_unique_account_id()` proc, also called by var-editing `add_to_accounts` if a duplicate exists?